subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/PrimaryInputDependencies.java

```diff
@@ -28,7 +28,7 @@
  * Attached to a property that should receive the <em>artifact dependencies</em> of the {@link PrimaryInput} of an artifact transform.
  *
  * The order of the files match that of the dependencies in the source artifact view.
- * The type of the injected dependencies is {@code Iterable<File>}.
+ * The type of the injected dependencies is {@link org.gradle.api.file.FileCollection}.
```

Comment:
It would be great to capture this programmatically. Like, we could add something like:

```java
@ProvidedParameterInjector(supportedTypes = {FileCollection.class})
public @interface PrimaryInputDependencies {
}
```

...and then we could automatically check if the target type is compatible. It would be a good way to document the fact that this is an injectable parameter annotation, too, so people could search for all the others.